docs/rfcs/1008-nightly-build.md

```diff
@@ -0,0 +1,208 @@
+  - [Slack notification (optional)](#slack-notification-optional)
+  - [Improve the unique dev format](#improve-the-unique-dev-format)
+- [F.A.Q](#faq)
+  - [Why do we need to release the nightly build? Or why we can't directly use the artifact link of a workflow?](#why-do-we-need-to-release-the-nightly-build-or-why-we-cant-directly-use-the-artifact-link-of-a-workflow)
+  - [Why do we use the `dev` identifier for a nightly build ?](#why-do-we-use-the-dev-identifier-for-a-nightly-build-)
+
+## Goals
+
+- Periodically test the release pipeline.
+- Have a recent build software at the ready (client and server).
+
+## What we already have
+
+We already have a workflow that create a release & build the artifact[^releaser-workflow] for that release.
+
+That workflow[^releaser-workflow] is triggered when we modify the workflow itself but notably when we push a tag on the repository that have the format `v[0-9]+.[0-9].[0-9]+*`.
+
+Currently, the workflow build the server python wheel for Linux, Windows & macOS.
+
+[^releaser-workflow]: https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/releaser.yml
+
+The script `releaser.py` can generate unique version that can be used for versioning and tagging nightly build.
+
+## What is missing
+
+From the release, we already have the server part indicated by:
+
+<https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/releaser.yml#L74-L79>
+
+### For the client part
+
+#### Missing version argument in the workflow `package-ionic-app.yml`
+
+The workflow ([`package-ionic-app.yml`]) need to be adapted like [`package-server.yml`] to take a version in argument.
+
+[`package-server.yml`]: https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/package-server.yml
+
+#### Include the client part in the releaser workflow
+
+We need to add the client to the workflow [`releaser.yml`].
+Thankfully we already have the workflow for that ([`package-ionic-app.yml`]).
+
+[`package-ionic-app.yml`]: https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/package-server.yml
+[`releaser.yml`]: https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/releaser.yml
+
+### Overwriting nightly release & automatic release
+
+Currently, the [`releaser.yml`] workflow create a [GitHub release] in draft mode.
+
+[GitHub release]: https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases
+
+This is problematic if the workflow create a `draft-release` for each nightly build (everyday so).
+
+> We would be overwhelmed by the amount of draft releases that would accumulate.
+
+To circumvent that we could make the workflow to remove the previous nightly release.
+But we have some limitation, the API to [delete a release] require a `release_id`.
```

Comment:
```suggestion
The [`releaser.yml`] workflow creates a [GitHub release] in draft mode. Creating a `draft-release` for _every_ nightly build (everyday so) is problematic because we might be overwhelmed by the number of draft releases.

We could [override the assets] of the previous nightly release (instead of recreating a new one). The problem with this approach is that GitHub sort the releases by creation date: nightly release will be put behind newer releases.

So the idea is to _remove_ the previous nigthly release and recreate a new one. In order to do so, the API to [delete a release] requires a `release_id`.

[override the assets]: https://stackoverflow.com/questions/62934246/github-update-overwrite-existing-asset-of-a-release
[GitHub release]: https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases
```